src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10289,6 +10329,51 @@ bool satisfiesConstraintChecks(MethodSymbol method)
 
                 return constraintsSatisfied;
             }
+
+            void populateExtensionTypeMethodsFromSingleBinder(
+                ExtensionScope scope,
+                MethodGroup methodGroup,
+                SyntaxNode node,
+                BoundExpression left,
+                string rightName,
+                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations)
+            {
+                Debug.Assert(left.Type is not null);
+
+                // PROTOTYPE need to confirm what we want for extension members on type parameters or values of type parameters
+                if (left.Type.IsTypeParameter())
+                    return;
+
+                int arity = typeArgumentsWithAnnotations.IsDefault ? 0 : typeArgumentsWithAnnotations.Length;
+
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(BindingDiagnosticBag.Discarded);
+                var lookupResult = LookupResult.GetInstance();
+
+                // Note: we only want members with matching arity
+                scope.Binder.LookupImplicitExtensionMembersInSingleBinder(
+                    lookupResult, left.Type, rightName, arity,
+                    basesBeingResolved: null, LookupOptions.Default, originalBinder: this, ref useSiteInfo);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    var members = ArrayBuilder<Symbol>.GetInstance();
+
+                    Symbol symbol = GetSymbolOrMethodOrPropertyGroup(lookupResult, node, rightName, arity, members,
+                        BindingDiagnosticBag.Discarded, wasError: out _, qualifierOpt: null);
+
+                    // Ignore non-method symbols
+                    if (symbol is null)
+                    {
+                        Debug.Assert(members.Count > 0);
+                        Debug.Assert(members[0].Kind is SymbolKind.Method);
+                        methodGroup.PopulateWithMethods(left, members, typeArgumentsWithAnnotations, isExtensionMethodGroup: false, resultKind: lookupResult.Kind);
```

Comment:
>PopulateWithMethods [](http://example.com/codeflow?start=36&length=19)

Is it safe to call this method multiple times? Is it going to discard results from the previous call?

Reply:
We call `methodGroup.Clear();` before calling `populateExtensionTypeMethodsFromSingleBinder`, so we're always populating into a clear method group.
The logic in `PopulateWithMethods` has assertions to that effect as well (only can call populate on a clear method group, see `VerifyClear()`).